test_conformance/basic/test_enqueued_local_size.cpp

```diff
@@ -74,6 +74,27 @@ test_enqueued_local_size(cl_device_id device, cl_context context, cl_command_que
     size_t localsize[2];
     int err;
 
+    // For a OpenCL-3.0 device this test should be skipped in the case that
+    // non-uniform work-groups are not supported.
+    if (get_device_cl_version(device) >= Version(3, 0))
+    {
+        cl_bool areNonUniformWorkGroupsSupported = false;
+        err = clGetDeviceInfo(device, CL_DEVICE_NON_UNIFORM_WORK_GROUP_SUPPORT,
+                              sizeof(areNonUniformWorkGroupsSupported),
+                              &areNonUniformWorkGroupsSupported, nullptr);
+        if (CL_SUCCESS != err)
+        {
+            test_error(err, "clGetDeviceInfo failed.");
+        }
+
+        if (CL_FALSE == areNonUniformWorkGroupsSupported)
+        {
+            log_info("Skipping since non-uniform work group sizes are not "
+                     "supported\n");
+            return TEST_SKIPPED_ITSELF;
```

Comment:
This is OK but it's a pretty big hammer since it means that none of this test will be executed without support for non-uniform work groups.  Would it instead be preferable to execute the test with uniform work groups on an OpenCL 3.0 device without non-uniform work group support so there is at least some coverage for `get_enqueued_local_size`?

Reply:
Thanks for the feedback @bashbaug. This is a really good point, I've updated the test slightly so that instead of skipping it clamps the global work dimensions such that the local work-group size divides the global size in the case that non-uniform work-groups are not supported. Let me know if this is the type of thing you had in mind?